rfcs/0013-avalanche.md

```diff
@@ -0,0 +1,98 @@
+
+**Cluster_info -> retransmit** = Used by TVUs to retransmit. Currently Layer-1 sends this to all tvu peers, leader is automatically excluded. This stage has bank and should be able to use it for weighted selection in the new design.
+
+**Cluster_info -> broadcast** = Used by leader to broadcast to layer-1 nodes. 
+
+**BroadcastStage -> run** = Used by leader (TPU) to broadcast to all validators. Currently all TVU Peers are considered layer-1. But blobs are transmitted sort of round robin. See Cluster_info->Broadcast. This stage doesn't have bank and will need it for weighted selection.
+
+
+## New Design
+
+                                     .-------------.
+                                     |             |
+                       .-------------+   Leader    +══════════════╗
+                       |             |             |              ║
+                       |             `-------------`              ║
+                       v                                          v
+                .-------------.                            .-------------.
+                |             +--------------------------->|             |
+           .----+ Validator 1 |                            | Validator 2 +═══╗
+           |    |             |<═══════════════════════════+             |   ║
+           |    `------+------`                            `------+------`   ║
+           |           |                                          ║          ║
+           |           `------------------------------.           ║          ║
+           |                                          |           ║          ║
+           |                     ╔════════════════════════════════╝          ║
+           |                     ║                    |                      ║
+           V                     v                    V                      v
+    .-------------.       .-------------.       .-------------.       .-------------.
+    |             |       |             |       |             |       |             |
+    | Validator 3 +------>| Validator 4 +══════>| Validator 5 +------>| Validator 6 |
+    |             |       |             |       |             |       |             |
+    `-------------`       `-------------`       `-------------`       `------+------`
+           ^                                                                 ║
+           ║                                                                 ║
+           ╚═════════════════════════════════════════════════════════════════╝
+
+#### What’s needed -
+The main goal of the new design is to enable Layer-2 nodes in the replication process. There are a few challenges with this and we should be able 
+to address some of them.  
+
+* Most importantly, there needs to be a mechanism to figure out which nodes belong in what layer of this avalanche. We need a Stake weighted selection for the nodes. This is missing and is very important. 
```

Comment:
@aeyakovenko, is using stake weight your direction? I think that unnecessarily mixes in leader rotation. The qualifier to be a layer-1 node is just that the node has a reputation for meeting all the performance metrics. That its network latency (from the leader, to its peers, and to the next level) is below certain thresholds. Previously, we've considered each fullnode maintaining a minimum spanning tree where the edges are network latency, and use that to caluculate positions in the Avalanche tree.

Reply:
Yeah it was @aeyakovenko's suggestion to use stake weights. I actually like it since it helps easily define boundaries between layers and layer-2 neighborhoods(assuming fanout is known and neighborhood size is known). It's likely all nodes have similar stake "knowledge" and can recreate the list of stake sorted tvus and figure out what their orderes are by checking that list. 

If we use a network latency based mechanism, each node might come up with a different picture of who its neighbors are and what layer it is in based on network latency. It won't be as straightforward to know if the network is being covered properly.

Reply:
Yeah, latency would be much less straightforward. I suppose we could start with stake and then do latency measurements to slash nodes as nonperforming. I can imagine that won't work well though, as a node with a very high stake could be physically very far from other nodes. That node would still get its normal shot at being leader, but it'd be burdensome for the network to rank it so high as a validator.

Reply:
Agreed. I'll go ahead with stake right now but we should be able to add _some_ network performance check for nodes in the future. Thanks to Erasure coding I think things might not be as bad as it might seem right now. @aeyakovenko mentioned something about maintaining QoS stats as well somewhere down the line. So I think we would eventually converge to something like you suggested.